apps/services/user-profile/src/app/user-profile/userProfile.controller.ts

```diff
@@ -76,9 +84,16 @@ export class UserProfileController {
     nationalId: string,
     @CurrentUser()
     user: User,
+    @Query('actorRequest') actorRequest?: boolean,
   ): Promise<UserProfile> {
-    if (nationalId != user.nationalId) {
-      throw new ForbiddenException()
+    if (actorRequest) {
```

Comment:
Another option would to make `locales` as sub-resource of `users`.

We have used `/me` and `/actor` as short hand for `/users/.nationalId`, where the `/me` uses `@CurrentUser()` while `/actor` uses `@CurrentActor()` (fallbacks to current user if no actor in token) decorators to get the corresponding nationalId. 

So we could create `GET /actor/locales` (ideally would return array of single locale as it is a collection endpoint) and then we could create `POST /actor/locales` to update the preferred locale.

I would be interested to hear what @eirikurn thinks about this.